Illegal field modifiers in class: 0x1A when using  io.quarkus.logging.Log in interface default method

### Describe the bug

We had a hard time to find out what is wrong due to not so helpful error message.

Tried to use io.quarkus.logging.Log inside a interface default method (Maybe this is generally a silly idea)

```
import io.quarkus.runtime.StartupEvent;
import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.event.Observes;


@ApplicationScoped
public class MyClass implements MyInterface {
    void onStart(@Observes StartupEvent ev) {
        log();
    }
}
```

```
import io.quarkus.logging.Log;

public interface MyInterface {
    default void log(){
        Log.info("log same stuff for all implementations");
    }
}
```


### Expected behavior

console log: `log same stuff for all implementations`

### Actual behavior

2022-11-04 16:50:49,262 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128)
	at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:347)
	at java.base/java.lang.Class.newInstance(Class.java:645)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:68)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:122)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	... 17 more
Caused by: java.lang.ClassFormatError: Illegal field modifiers in class org/acme/MyInterface: 0x1A
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:490)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:450)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:490)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:450)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at org.acme.MyClass_Bean.<init>(Unknown Source)
	at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans1(Unknown Source)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Unknown Source)
	at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:127)
	at io.quarkus.arc.Arc.initialize(Arc.java:31)
	at io.quarkus.arc.runtime.ArcRecorder.initContainer(ArcRecorder.java:43)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources1337887788.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources1337887788.deploy(Unknown Source)
	... 18 more

2022-11-04 16:50:49,263 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure

### How to Reproduce?

See codeblocks in the description, as it is so few code necessary. If it is helpful I can also push a reproducer.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18 OpenJDK Runtime Environment Temurin-17.0.2+8 (build 17.0.2+8) OpenJDK 64-Bit Server VM Temurin-17.0.2+8 (build 17.0.2+8, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingWithPanacheProcessor.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingWithPanacheProcessor.java
@@ -43,7 +43,8 @@ public class LoggingWithPanacheProcessor {
     /**
      * Makes the following modifications to the visited class:
      * <ul>
-     * <li>adds a {@code private static final} field of type {@code org.jboss.logging.Logger};</li>
+     * <li>adds a {@code private static final} field of type {@code org.jboss.logging.Logger}
+     * ({@code public} in case the class is an interface, to obey the JVMS rules);</li>
      * <li>initializes the field (to {@code Logger.getLogger(className)}) at the beginning of the
      * static initializer (creating one if missing);</li>
      * <li>rewrites all invocations of {@code static} methods on {@code io.quarkus.logging.Log}
@@ -57,6 +58,8 @@ public class LoggingWithPanacheProcessor {
         private final String className;
         private final String classNameBinary;
 
+        private boolean isInterface;
+
         private boolean generatedLoggerField;
         private boolean generatedLoggerFieldInitialization;
 
@@ -64,6 +67,14 @@ public class LoggingWithPanacheProcessor {
             super(Gizmo.ASM_API_VERSION, visitor);
             this.className = className;
             this.classNameBinary = className.replace(".", "/");
+        }
+
+        @Override
+        public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+            super.visit(version, access, name, signature, superName, interfaces);
+            if ((access & Opcodes.ACC_INTERFACE) != 0) {
+                isInterface = true;
+            }
         }
 
         @Override
@@ -190,8 +201,11 @@ public class LoggingWithPanacheProcessor {
         }
 
         private void generateLoggerField() {
-            super.visitField(Opcodes.ACC_PRIVATE | Opcodes.ACC_STATIC | Opcodes.ACC_FINAL,
-                    SYNTHETIC_LOGGER_FIELD_NAME, JBOSS_LOGGER_DESCRIPTOR, null, null);
+            // interface fields must be public static final per the JVMS
+            int access = isInterface
+                    ? Opcodes.ACC_PUBLIC | Opcodes.ACC_STATIC | Opcodes.ACC_FINAL
+                    : Opcodes.ACC_PRIVATE | Opcodes.ACC_STATIC | Opcodes.ACC_FINAL;
+            super.visitField(access, SYNTHETIC_LOGGER_FIELD_NAME, JBOSS_LOGGER_DESCRIPTOR, null, null);
             generatedLoggerField = true;
         }
     }

--- a/integration-tests/logging-panache/src/test/java/io/quarkus/logging/LoggingBean.java
+++ b/integration-tests/logging-panache/src/test/java/io/quarkus/logging/LoggingBean.java
@@ -4,7 +4,7 @@ import javax.annotation.PostConstruct;
 import javax.inject.Singleton;
 
 @Singleton
-public class LoggingBean {
+public class LoggingBean implements LoggingInterface {
     // not final to prevent constant inlining
     private static String msg = "Heya!";
 
@@ -18,6 +18,8 @@ public class LoggingBean {
     }
 
     public void doSomething() {
+        inheritedMethod("abc");
+
         if (Log.isDebugEnabled()) {
             Log.debug("starting massive computation");
         }

--- a/integration-tests/logging-panache/src/test/java/io/quarkus/logging/LoggingInterface.java
+++ b/integration-tests/logging-panache/src/test/java/io/quarkus/logging/LoggingInterface.java
@@ -1,0 +1,9 @@
+package io.quarkus.logging;
+
+public interface LoggingInterface {
+    default void inheritedMethod(String param) {
+        if (Log.isInfoEnabled()) {
+            Log.infof("Default method from interface: %s", param);
+        }
+    }
+}

--- a/integration-tests/logging-panache/src/test/java/io/quarkus/logging/LoggingWithPanacheTest.java
+++ b/integration-tests/logging-panache/src/test/java/io/quarkus/logging/LoggingWithPanacheTest.java
@@ -17,8 +17,8 @@ import io.quarkus.test.QuarkusUnitTest;
 public class LoggingWithPanacheTest {
     @RegisterExtension
     static final QuarkusUnitTest test = new QuarkusUnitTest()
-            .withApplicationRoot((jar) -> jar
-                    .addClasses(LoggingBean.class, LoggingEntity.class, NoStackTraceTestException.class))
+            .withApplicationRoot((jar) -> jar.addClasses(LoggingBean.class, LoggingInterface.class, LoggingEntity.class,
+                    NoStackTraceTestException.class))
             .overrideConfigKey("quarkus.log.category.\"io.quarkus.logging\".min-level", "TRACE")
             .overrideConfigKey("quarkus.log.category.\"io.quarkus.logging\".level", "TRACE")
             .setLogRecordPredicate(record -> record.getLoggerName().startsWith("io.quarkus.logging.Logging"))
@@ -29,6 +29,7 @@ public class LoggingWithPanacheTest {
                 assertThat(lines).containsExactly(
                         "[INFO] Heya!",
                         "[TRACE] LoggingBean created",
+                        "[INFO] Default method from interface: abc",
                         "[DEBUG] starting massive computation",
                         "[DEBUG] one: 42",
                         "[TRACE] two: 42 | 13",

